### PR TITLE
Private routing, linked "Sign Up" button to endpoint, data validation for Login and Signup

### LINKED ISSUE #82: Link 'Sign Up' button on landing page to endpoint
**User Story**
As a user, I want to be able to create an account using the signup form on the landing page

**Acceptance Criteria**
* If email address & password are appropriate, then a new account is created
* If new account is created, user is logged in automatically & taken to home page
* If email and/or password are not appropriate, an error message is displayed

**Additional Context**
The landing page has already been designed and the signup form has been created. This issue only involves sending the form data to the appropriate API endpoint for signups. 

**Dependencies**
#52

### LINKED ISSUE #96: Add database validation for Firebase API
**User Story**
Ensure user data integrity.

**Acceptance Criteria**
- Data-checking mechanisms are present in POST and PUT methods to ensure entries and changes to database are valid.
- Include test cases which taste abnormal data cases.

**Dependencies**
- Jest is required to run unit tests with mock database.

### PR DESCRIPTION
**Pull request by lucas2005gao**
Issue:
#96
#82

Solution:
Private routing

Wrapped Home page in a private route that uses authContext
Link "Sign Up" button to endpoint
Data validation for Firebase API
Express validator used to check email and password data for Login and Signup

### CODE REVIEW

**Review by max-gurr — APPROVED** [review]
Have seen this working, so am happy to approve!

File: client/src/pages/LandingPage/index.js [hunk 1]
```diff
@@ -34,6 +32,67 @@ const useStyles = makeStyles({
...
     const classes = useStyles();
     const history = useHistory();
+    const [signupMsgboxOpen, setSignupMsgboxOpen] = useState(false);
+    const [signupMsg, setSignupMsg] = useState("Sign up failed, invalid Email address or password");
+    const [email, setEmail] = useState("");
+    const [password, setPassword] = useState("");
+
+    const fetchSignup = (username, password) => {
+        console.log(username);
+        console.log(password);
```
Comment (franc-hoon) [inline client/src/pages/LandingPage/index.js]: remove? + all other console logs in this page

**Review by franc-hoon — CHANGES_REQUESTED** [review]
looks good, just need to remove console logs. Also what's the setToken for in LoginPage/index.js?

**Review by gsin387 — APPROVED** [review]
Looks good, not sure if the package-lock.json file should be pushed tho.

**Review by DanielDowling999 — APPROVED** [review]
Agreed about the removing of console logs (putting the inputted username and password out on the console is quite bad lmao), but otherwise everything looks good!

**Review by franc-hoon — APPROVED** [review]
lgtm

File: client/src/pages/LandingPage/index.js (hunk 1 above)
Comment (max-gurr) [reply client/src/pages/LandingPage/index.js]: The setToken is to store the token in localstorage through the ```use-persisted-state``` package. This is how the auth context uses it for private routing

**Pull request merged**